tests/src/lib.rs

```diff
@@ -246,6 +248,136 @@ where
     RoundtripResult::Ok(buf1)
 }
 
+pub enum RoundtripInput<'a> {
+    Protobuf(&'a [u8]),
+    Json(&'a str),
+}
+
+pub enum RoundtripOutputType {
+    Protobuf,
+    Json,
+}
+
+pub enum RoundtripResult2 {
+    /// The roundtrip to protobuf succeeded.
+    Protobuf(Vec<u8>),
+    /// The roundtrip to json succeeded.
+    Json(String),
+    /// The data could not be encoded. This could indicate a bug in prost,
+    /// or it could indicate that the data was invalid (eg. violating message invariants).
+    EncodeError(anyhow::Error),
+    /// The data could not be decoded. This could indicate a bug in prost,
+    /// or it could indicate that the input was bogus.
+    DecodeError(anyhow::Error),
+    /// Re-encoding or validating the data failed.  This indicates a bug in `prost`.
+    Error(anyhow::Error),
+}
+
+pub fn roundtrip2<M>(
```

Comment:
Why is a new roundtrip function needed? Can the original roundtrip be improved?

Reply:
This is just a thing that I haven't cleaned up yet. The new function is intended to replace the old one, I just haven't gotten to replace all the uses of the old one yet 😅.